File: 2019W18/PreppindataW18.py
import pandas as pd

xls = pd.ExcelFile(r"E:\anime.xlsx")

anime = pd.read_excel(xls,"anime") 
anime.drop(['episodes'],axis=1,inplace = True)
temp = anime[((anime['type'] == 'TV') | (anime['type'] == 'Movie')) & (anime['members'] >= 10000)].dropna().copy()
genre_split = pd.DataFrame([x for x in temp['genre'].str.split(',').values.tolist()])

final = pd.concat([temp[['name','genre','type','rating','members']].reset_index(drop=True), genre_split], axis=1, sort=False)
final.drop(['genre'],axis=1,inplace = True)
final = pd.melt(final, id_vars=['name','type','rating','members'], value_name='Genre')
final['Genre'] = final['Genre'].str.strip()
final.drop(['variable'],axis=1,inplace = True)
final=final.dropna()
final['Avg Rating'] = round((final.groupby(['Genre','type']))['rating'].transform('mean'),2)
final['Avg Viewers'] = round((final.groupby(['Genre','type']))['members'].transform('mean'))
final = final.groupby(['Genre','type']).apply(lambda t: t[t.rating==t.rating.max()])

final.drop(['members'],axis=1,inplace = True)
final.rename(columns={'type': 'Type', 'name': 'Prime Example', 'rating' : 'Max Rating'}, inplace=True)
final = final[['Genre','Type','Avg Rating','Max Rating','Avg Viewers','Prime Example']].reset_index(drop=True)

File: 2019W18/preppindataw18.R
library(readxl)
library(dplyr)
library(tidyr)

anime <- read_excel("E:/anime.xlsx", sheet = "anime", 
                    col_types = c("text", "text", "text", 
                                  "text", "text", "numeric", "numeric")) %>%
  filter((`type`=='Movie' | `type`=='TV') & `members`>=10000) %>%
  drop_na() 

max_genre <- as.numeric(unlist(lapply(anime %>% select('genre'), function(x) max(sapply(strsplit(x, ','), length)))))

final <-  anime  %>%
  separate(`genre`, paste0("genre_split_", seq(1:max_genre)), sep = ',') %>%
  select(-c('anime_id','episodes')) %>%
  gather(., key="var", value="Genre", -c('name','type','rating','members')) %>%
  mutate('Genre' = trimws(`Genre`)) %>%
  select(-("var")) %>%
  drop_na() %>%
  group_by(`Genre`,`type`) %>%
  mutate('Avg Rating' = mean(`rating`),
         'Avg Viewers' = mean(`members`),
         'Max Rating' = max(`rating`)) %>%
  filter(`rating`==`Max Rating`) %>%
  rename('Type' = 'type',
         'Prime Example' = 'name') %>%
  select(c('Genre','Type','Avg Rating','Max Rating','Avg Viewers','Prime Example'))

View(final)